doc/developer/design/20230306_persist_mfp_pushdown.md

```diff
@@ -0,0 +1,272 @@
+- Feature name: Persist Filter Pushdown
+- Associated: https://github.com/MaterializeInc/materialize/issues/12684
+
+# Summary
+[summary]: #summary
+
+Queries against large collections in Materialize can be very expensive:
+readers generally need to fetch all the data in the collection as of the query time,
+even if they're only interested in a small subset.
+By pushing down filters into the Persist source,
+we can dramatically improve performance for certain subsets of queries,
+especially those that filter down to a narrow window of time.
+
+# Motivation
+[motivation]: #motivation
+
+Unlock append-only sources for customers via temporal filters. Right now, using an append-only `SOURCE` in Materialize is painful because `MATERIALIZED VIEW` restart times increase without bound as data accumulates. However, if there is a [temporal filter](https://materialize.com/docs/sql/patterns/temporal-filters/) to bound the size of the dataflow's input data (e.g. the last 30 days of data), we may be able to use the described optimization to similarly bound the restart times.
+
+This technique is not limited to temporal filters or sources. In fact, the optimization will be picked up anywhere we can apply it. However, we're hoping to focus on polishing the subset relevant to append-only sources in the first version of this shipped to customers.
+
+# Explanation
+[explanation]: #explanation
+
+## Schemas
+
+A `Schema` trait will be introduced to map data back and forth between Persist columns. Any Rust type
+that is to roundtrip through Persist must have a corresponding `Schema`.
```

Comment:
nit: wrapping on this doc is all over the place now, let's pick a width and align everything to that

Reply:
my two cents: It makes it easier to align a review comment to a specific sentence if each sentence in a paragraph starts on a new line.